src/test/java/seedu/address/logic/commands/AddCommandTest.java

```
@@ -0,0 +1,170 @@
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.ReadOnlyPerson;
+import seedu.address.model.person.UniquePersonList.DuplicatePersonException;
+import seedu.address.model.person.UniquePersonList.PersonNotFoundException;
+import seedu.address.model.tag.UniqueTagList;
+import seedu.address.testutil.TestPerson;
+import seedu.address.testutil.TypicalTestPersons;
+
+public class AddCommandTest {
+
+    private AddModelStub model;
+
+    private TestPerson newPerson;
+    private TestPerson duplicatePerson;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Before
+    public void setUp() throws Exception {
+        TypicalTestPersons typicalTestPersons = new TypicalTestPersons();
+
+        newPerson = typicalTestPersons.alice;
+        duplicatePerson = typicalTestPersons.daniel;
+        model = new AddModelStub(duplicatePerson);
+    }
+
+    @Test
+    public void constructor_invalidName_throwsIllegalValueException() throws Exception {
+        thrown.expect(IllegalValueException.class);
+        new AddCommand("", "61239876", "valid.email@mail.com", "valid address", new HashSet<>());
+    }
+
+    @Test
+    public void constructor_invalidPhone_throwsIllegalValueException() throws Exception {
+        thrown.expect(IllegalValueException.class);
+        new AddCommand("valid name", "", "valid.email@mail.com", "valid address", new HashSet<>());
+    }
+
+    @Test
+    public void constructor_invalidEmail_throwsIllegalValueException() throws Exception {
+        thrown.expect(IllegalValueException.class);
+        new AddCommand("valid name", "61239876", "", "valid address", new HashSet<>());
+    }
+
+    @Test
+    public void constructor_invalidAddress_throwsIllegalValueException() throws Exception {
+        thrown.expect(IllegalValueException.class);
+        new AddCommand("valid name", "61239876", "valid.email@mail.com", "", new HashSet<>());
```

Hmm... I agree that it can feel quite repetitive. However the purpose of these tests is to check that the correct Exceptions are thrown, rather than actually testing the logic of processing invalid values. (which are covered in tests for `Address`, `Email`, etc in PR #253, and has more variations in invalid values)